pkg/operator/ceph/cr_manager.go

```diff
@@ -52,7 +71,93 @@ var (
 	webhookResources = []webhook.Validator{&cephv1.CephCluster{}, &cephv1.CephBlockPool{}, &cephv1.CephObjectStore{}}
 )
 
-func (o *Operator) startManager(context context.Context, namespaceToWatch string, mgrErrorCh chan error) {
+var (
+	// EnableMachineDisruptionBudget checks whether machine disruption budget is enabled
+	EnableMachineDisruptionBudget bool
+)
+
+// AddToManagerFuncsMaintenance is a list of functions to add all Controllers to the Manager (entrypoint for controller)
+var AddToManagerFuncsMaintenance = []func(manager.Manager, *controllerconfig.Context) error{
+	clusterdisruption.Add,
+}
+
+// MachineDisruptionBudgetAddToManagerFuncs is a list of fencing related functions to add all Controllers to the Manager (entrypoint for controller)
+var MachineDisruptionBudgetAddToManagerFuncs = []func(manager.Manager, *controllerconfig.Context) error{
+	machinelabel.Add,
+	machinedisruption.Add,
+}
+
+// AddToManagerFuncs is a list of functions to add all Controllers to the Manager (entrypoint for controller)
+var AddToManagerFuncs = []func(manager.Manager, *clusterd.Context) error{
+	crash.Add,
+	pool.Add,
+	objectuser.Add,
+	realm.Add,
+	zonegroup.Add,
+	zone.Add,
+	object.Add,
+	file.Add,
+	nfs.Add,
+	rbd.Add,
+	client.Add,
+	mirror.Add,
+}
+
+// AddToManagerOpFunc is a list of functions to add all Controllers to the Manager (entrypoint for
+// controller)
+var AddToManagerOpFunc = []func(manager.Manager, *clusterd.Context, opcontroller.OperatorConfig) error{
+	Add,
+	csi.Add,
+	agent.Add,
+}
+
+// AddToManager adds all the registered controllers to the passed manager.
+// each controller package will have an Add method listed in AddToManagerFuncs
+// which will setup all the necessary watch
+func (o *Operator) addToManager(m manager.Manager, c *controllerconfig.Context) error {
+	if c == nil {
+		return errors.New("nil context passed")
+	}
+
+	// Run CephCluster CR
+	if err := cluster.Add(m, c.ClusterdContext, o.clusterController); err != nil {
+		return err
+	}
+
+	// Add Ceph child CR controllers
+	for _, f := range AddToManagerFuncs {
+		if err := f(m, c.ClusterdContext); err != nil {
+			return err
+		}
+	}
+
+	// Add maintenance controllers
+	for _, f := range AddToManagerFuncsMaintenance {
+		if err := f(m, c); err != nil {
+			return err
+		}
+	}
+
+	// If machine disruption budget is enabled let's add the controllers
+	if EnableMachineDisruptionBudget {
```

Comment:
On a separate topic, I wonder if anybody ever enables the machine disruption budgets and this should be deprecated...

Reply:
Agreed...